core/modules/qdisp/Executive.cc

```diff
@@ -407,25 +427,25 @@ void Executive::_setup() {
 
 /// Check to see if a requester should retry, based on how many retries have
 /// been attempted. Increments the retry counter as a side effect.
-bool Executive::_shouldRetry(int refNum) {
+bool Executive::_shouldRetry(int jobId) {
     const int MAX_RETRY = 5;
     std::lock_guard<std::mutex> lock(_retryMutex);
-    IntIntMap::iterator i = _retryMap.find(refNum);
+    IntIntMap::iterator i = _retryMap.find(jobId);
     if(i == _retryMap.end()) {
-        _retryMap[refNum] = 1;
+        _retryMap[jobId] = 1;
     } else if(i->second < MAX_RETRY) {
-        _retryMap[refNum] = 1 + i->second;
+        _retryMap[jobId] = 1 + i->second;
     } else {
         return false;
     }
     return true;
 }
```

Comment:
:+1:, I've added a boolean should_retry, hope it's fine now. I didn't write this code, so I can't fine tune-it in a deeper level quickly.